### PR TITLE
FIX-#2249: changed 'return' -> 'assert' in df_equals

### LINKED ISSUE #2249: df_equals do not assert frame missmatches in some cases
Sometimes `df_equals` just returns `False` when frame aren't equals and not asserting it
https://github.com/modin-project/modin/blob/dd691cd0d4bb5403b38242563ed8b6ddb316734a/modin/pandas/test/utils.py#L506-L510

### PR DESCRIPTION
**Pull request by dchigarev**
Signed-off-by: Dmitry Chigarev <dmitry.chigarev@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #2249 
- [x] tests passing

### CODE REVIEW

File: modin/pandas/test/utils.py [hunk 1]
```diff
@@ -505,9 +505,9 @@ def df_equals(df1, df2):
 
     if isinstance(df1, pandas.DataFrame) and isinstance(df2, pandas.DataFrame):
         if (df1.empty and not df2.empty) or (df2.empty and not df1.empty):
-            return False
+            assert False
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:508]: @dchigarev please add message

File: modin/pandas/test/utils.py [hunk 2]
```diff
@@ -505,9 +505,9 @@ def df_equals(df1, df2):
...
         elif df1.empty and df2.empty and type(df1) != type(df2):
-            return False
+            assert False
```
Comment (anmyachev) [inline modin/pandas/test/utils.py:510]: same

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: modin/pandas/test/utils.py (hunk 1 above)
Comment (dchigarev) [reply modin/pandas/test/utils.py]: done

File: modin/pandas/test/utils.py (hunk 2 above)
Comment (dchigarev) [reply modin/pandas/test/utils.py]: done

**Review by anmyachev — APPROVED** [review]
LGTM!

**Pull request merged**